plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcPlugin.java

```diff
@@ -47,9 +64,84 @@ public void testRuleBasedIdentifierCanBeUsedTogetherWithCacheBased()
                 .shutdown();
     }
 
+    @RepeatedTest(100)
+    void testConfigurationDoesNotLeakBetweenCatalogs()
+    {
+        TestingJdbcPlugin plugin = new TestingJdbcPlugin("test_jdbc", TestingJdbcModule::new);
+        ConnectorFactory connectorFactory = getOnlyElement(plugin.getConnectorFactories());
+
+        try (ExecutorService executor = Executors.newFixedThreadPool(2)) {
+            Future<Connector> pushDownEnabledFuture = executor.submit(() -> connectorFactory.create(
+                    TestingJdbcModule.CATALOG_WITH_PUSH_DOWN_ENABLED,
+                    ImmutableMap.of("connection-url", createH2ConnectionUrl(), "join-pushdown.enabled", "true"),
+                    new TestingConnectorContext()));
+            Future<Connector> pushDownDisabledFuture = executor.submit(() -> connectorFactory.create(
+                    TestingJdbcModule.CATALOG_WITH_PUSH_DOWN_DISABLED,
+                    ImmutableMap.of("connection-url", createH2ConnectionUrl(), "join-pushdown.enabled", "false"),
+                    new TestingConnectorContext()));
+
+            AtomicReference<Connector> catalogWithPushDownEnabled = new AtomicReference<>();
+            AtomicReference<Connector> catalogWithPushDownDisabled = new AtomicReference<>();
+            assertThatCode(() -> {
+                catalogWithPushDownEnabled.set(pushDownEnabledFuture.get());
+                catalogWithPushDownDisabled.set(pushDownDisabledFuture.get());
+            }).doesNotThrowAnyException();
+
+            catalogWithPushDownEnabled.get().shutdown();
+            catalogWithPushDownDisabled.get().shutdown();
```

Comment:
No need for `AtomicReference` and `doesNotThrowAnyException`

given that those futures are completed concurrently and we do nothing with their result, this would be suffice

```java
pushDownEnabledFuture.get().shutdown();
pushDownDisabledFuture.get().shutdown();
```